cmd/goal/clerk.go

```diff
@@ -663,7 +682,7 @@ var signCmd = &cobra.Command{
 				signedTxn = uncheckedTxn
 			} else {
 				// sign the usual way
-				signedTxn, err = client.SignTransactionWithWallet(wh, pw, uncheckedTxn.Txn)
+				signedTxn, err = client.SignTransactionWithWalletAndSigner(wh, pw, signerAddress, uncheckedTxn.Txn)
```

Comment:
Don't we need some type of input validation for `signerAddress` ?

Reply:
Yeah, I wasn't sure whether the libgoal function should take in an Address or a string. I made it a string because that's what libgoal's `MultisigSignTransactionWithWallet` function takes in for its signer address, but I could switch it and have the address decoding in goal instead of libgoal if you think that would be better.